index.html

```diff
@@ -2209,218 +2270,232 @@ <h3><dfn>Writing or pushing content</dfn></h3>
+                </li>
+                <li>
+                  Set `this`.[[\PushMessage]] to |output|.
+                </li>
+                <li>
+                  Set <a>pending push tuple</a> to (`this`, |p|).
+                </li>
+                <li>
+                  Run the <a>start the push timer</a> steps.
+                </li>
+                <li>
+                  Run the <a>add write handler</a> steps.
+                </li>
+                <p class="note">
+                  If <a>NFC is suspended</a>,
+                  continue waiting until |timer| expires (if set), or
+                  promise is aborted by the user, or until an <a>NFC device</a>
+                  comes within communication range.
+                </p>
+              </ol>
+            </li>
+          </li>
+        </ol>
+      </p>
+
+      <p id="steps-push-timer">
+        To <dfn>start the push timer</dfn>, run these steps:
+        <ol class=algorithm>
+          <p>
+            Let |p:Promise| be the <a>pending push tuple</a>'s promise.
+          </p>
+          <li>
+            Let |writer| be the <a>pending push tuple</a>'s writer.
+          </li>
+          <li>
+            Let |timeout:number| be |writer|.[[\PushOptions]]'s timeout.
+            <ul>
               <li>
                 If |timeout| value is `NaN` or negative, reject
-                |p| with a {{TypeError}} and abort
-                these steps.
+                |p| with a {{TypeError}} and abort these steps.
               </li>
               <li>
                 If |timeout| value is not supported by the UA, reject
                 |p| with a {{"NotSupportedError"}}
                 {{DOMException}} and abort these steps.
               </li>
               <li>
-                Let |output| be the notation for the <a>NDEF message</a>
-                to be created by UA, as the result of passing
-                |message| to <a>create Web NFC message</a>.
-                If this throws an exception, reject |p| with that
-                exception and abort these steps.
-              </li>
-              <li>
-                If |target| is "`any`", run the following
-                steps twice, once with |slot| set to the value
-                "`tag`", and once set to the value "`peer`";
-                otherwise run the following step once, with
-                |slot| set to the value of |target|.
-              </li>
-              <li>
-                Attempt to <a>abort a pending push operation</a>.
+                If |timeout| value is not equal to `Infinity`, abort these
+                steps.
               </li>
+            </ul>
+          </li>
+          <li>
+            Start a timer |timer| with the timeout value set to
+            |timeout| and set |writer|.[[\PushTimer]] to |timer|.
+          </li>
+          <li>
+            If |timer| expires, reject |p| with {{"TimeoutError"}}
```

Comment:
when?

Reply:
Either is fine for me, "if" seemed more unambiguous. "When" might imply that it _will_ happen, the only question is when.

Reply:
If it expires...  Doesn't mention when to act.. if I know it will expire I can act immediately, which makes no sense